runtime/frequency/src/lib.rs

```diff
@@ -146,17 +149,210 @@ pub type Executive = frame_executive::Executive<
+				Self::join_keys(&SUDO_PREFIX, &PALLET_VERSION_PREFIX);
+
+			let mut keys_exist = frame_support::storage::unhashed::exists(sudo_key.as_slice());
+			log::info!("❓       Sudo Key storage exists ===>  {:?}", keys_exist);
+			keys_exist = frame_support::storage::unhashed::exists(sudo_pallet_version.as_slice());
+			log::info!("❓       Sudo PalletVersion storage exists ===>  {:?}", keys_exist);
+			keys_exist
+		}
+
+		#[cfg(not(feature = "frequency"))]
+		fn transfer_sudo_balance_to_treasury() {
+			log::warn!(
+				"‼️        transfer_sudo_balance_to_treasury was called but should not have been"
+			);
+		}
+
+		// TODO: correct weight
+		fn weights_from(reads: u64, writes: u64) -> Weight {
+			Weight::from_ref_time(0u64)
+				.saturating_add(RocksDbWeight::get().reads(reads))
+				.saturating_add(RocksDbWeight::get().writes(writes))
+		}
+
+		fn remove_storage(key: &Vec<u8>) {
+			frame_support::storage::unhashed::kill(key.as_slice());
+		}
+	}
+
+	impl OnRuntimeUpgrade for RemoveSudo {
+		// on_runtime_upgrade is the only OnRuntimeUpgrade trait function that must be defined for all configs
+		// do nothing if we are not on mainnet.
+		#[cfg(not(feature = "frequency"))]
+		fn on_runtime_upgrade() -> Weight {
+			Weight::zero()
+		}
+
+		// Do this if we are on mainnet
+		#[cfg(feature = "frequency")]
+		fn on_runtime_upgrade() -> Weight {
+			// TODO: move storage removal to fn in struct above
+
+			// keep track of reads/writes
+			let mut reads: u64 = 4; // from the two calls to keys_exist
+			let mut writes: u64 = 0;
+
+			if !Self::keys_exist() {
+				// we don't want to proceed if there is even a partial migration.
+				log::warn!("Sudo Storage Migration run on already migrated database. This migration should be removed.");
+			} else {
+				let sudo_key: Vec<u8> = Self::join_keys(&SUDO_PREFIX, &KEY_PREFIX);
+				// get the value out so we can transfer the funds later.
+				let storage_result: Option<AccountId> =
+					frame_support::storage::unhashed::get(sudo_key.as_slice());
+
+				let sudo_pallet_version: Vec<u8> =
+					Self::join_keys(&SUDO_PREFIX, &PALLET_VERSION_PREFIX);
+
+				Self::remove_storage(&sudo_key);
+				Self::remove_storage(&sudo_pallet_version);
+				writes += 2;
+
+				// "To ensure that this function results in a killed account, you might need to prepare the account by
+				// removing any reference counters, storage deposits, etc…"
+				// https://paritytech.github.io/substrate/master/pallet_balances/pallet/struct.Pallet.html#method.transfer_all
+				// Theoretical "ensure" with emphasis on "might"...
+				if storage_result.is_some() {
+					let from: AccountId = storage_result.unwrap();
+					// TODO: should extrinsic call be included in the weight? Surely it is added by the txn call.
+					Self::transfer_sudo_balance_to_treasury(from);
+					reads += 1;
+					writes += 1;
+				}
+				System::deposit_event(frame_system::Event::CodeUpdated);
+			}
+			let _ = Self::keys_exist();
```

Comment:
Why do we run this again?

Reply:
THis is only run in try-runtime, and it just checks for the keys' existence, and logs whether they are still there.